pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -0,0 +1,160 @@
+	// Create a new controller
+	c, err := controller.New("nodenetworkconfigurationpolicy-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource NodeNetworkConfigurationPolicy
+	err = c.Watch(&source.Kind{Type: &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}}, &handler.EnqueueRequestForObject{}, forThisNodePredicate(mgr.GetClient()))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileNodeNetworkConfigurationPolicy implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileNodeNetworkConfigurationPolicy{}
+
+// ReconcileNodeNetworkConfigurationPolicy reconciles a NodeNetworkConfigurationPolicy object
+type ReconcileNodeNetworkConfigurationPolicy struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a NodeNetworkConfigurationPolicy object and makes changes based on the state read
+// and what is in the NodeNetworkConfigurationPolicy.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNodeNetworkConfigurationPolicy) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling NodeNetworkConfigurationPolicy")
+
+	// Fetch the NodeNetworkConfigurationPolicy instance
+	instance := &nmstatev1alpha1.NodeNetworkConfigurationPolicy{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	nodeNetworkState := &nmstatev1alpha1.NodeNetworkState{}
+	nodeNetworkStateKey := types.NamespacedName{Name: nodeName}
+	err = r.client.Get(context.TODO(), nodeNetworkStateKey, nodeNetworkState)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Info(fmt.Sprintf("the NodeNetworkState for %s is not there yet, let's requeue", nodeName))
+			// If there is no NodeNetworkState let's requeue could be that
+			// we are in the middle of the creation
+			return reconcile.Result{Requeue: true}, nil
+		}
+		// Error reading the nodeNetworkState - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// FIXME: We have to merge it somehow in case of multiple policies applied
```

Comment:
please add TODO for Patch operation instead of Update. Otherwise, we will end up without observedState and conditions for some time.

Reply:
Done